custom_dc/custom_data.md

```diff
@@ -0,0 +1,291 @@
+
+The variable values must be numeric. Zeros and null values are accepted: zeros will be recorded and null values ignored.
+
+All headers must be in camelCase.
+
+The following are some valid examples of headers:  
+   
+```csv 
+geoId,observationYear,statVar1,statVar2  
+06,2021,555,666  
+08,2021,10,10  
+```  
+```csv 
+name,observationYear,statVar1,statVar2  
+California,2021,555,666  
+Colorado,2021,10,10  
+```  
+```csv 
+dcId,observationYear,statVar1,statVar2  
+geoId/06,2021,555,666  
+geoId/08,2021,10,10  
+```
+
+## Write the data config file
+
+The config.json file specifies how the CSV contents should be mapped and resolved to the Data Commons schema. See the example in the [`sample/config.json`](https://github.com/datacommonsorg/website/blob/master/custom_dc/sample/config.json) file provided, which describes the data in the [`sample/average_annual_wage.csv`](https://github.com/datacommonsorg/website/blob/master/custom_dc/sample/average_annual_wage.csv) and [`sample/gender_wage_gap.csv`](https://github.com/datacommonsorg/website/blob/master/custom_dc/sample/gender_wage_gap.csv) files.  
+
+Here is the general spec for the JSON file:  
+
+<pre>
+{  
+  "inputFiles": {  
+    "<var>FILE_NAME1</var>": {  
+      "entityType": "<var>ENTITY_PROPERTY</var>",  
+      "ignoreColumns": ["<var>COLUMN1</var>", "<var>COLUMN2</var>", ...],  
+      "provenance": "<var>NAME</var>"  
+    },  
+    "<var>FILE_NAME2</var>": {  
+     ...  
+    },  
+ ...  
+  "variables": {  
+    "<var>VARIABLE1</var>": {"group": "<var>GROUP_NAME1</var>"},  
+    "VARIABLE2": {"group": "<var>GROUP_NAME1</var>"},  
+    "<var>VARIABLE3</var>": {  
+      "name": "<var>DISPLAY_NAME</var>",  
+      "description": "<var>DESCRIPTION</var>",  
+      "searchDescriptions": ["<var>SENTENCE1</var>", "<var>SENTENCE2</var>", ...],  
+      "group": "<var>GROUP_NAME2</var>",  
+      "properties": {  
+        "<var>PROPERTY_NAME1</var>":"<var>VALUE</var>",  
+        "<var>PROPERTY_NAME2</var>":"<var>VALUE</var>",  
+         …  
+           }  
+    },  
+  },   
+  "sources": {  
+    "<var>SOURCE_NAME1</var>": {  
+      "url": "<var>URL</var>",  
+      "provenances": {  
+        "<var>PROVENANCE_NAME1</var>": "<var>URL</var>",  
+        "<var>PROVENANCE_NAME2</var>": "<var>URL</var>",  
+        ...  
+      }  
+    }  
+  }  
+}  
+</pre>
+
+Each section contains some required and optional fields, which are described in detail below.
+
+### Input files
+
+The top-level `inputFiles` field should encode a map from the input file name to parameters specific to that file. Keys can be individual file names or wildcard patterns if the same config applies to multiple files.
+
+You can use the `*` wildcard; matches are applied in the order in which they are specified in the config. For example, in the following:  
+
+```json
```

Comment:
nit: Probably can use the format `javascript` or can be left blank since the `...` in the syntax is not valid json or js syntax.

Reply:
OK, done.